designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtype/ProcessingTypeData.scala

```diff
@@ -97,20 +101,24 @@ object ProcessingTypeData {
       deploymentManagerProvider: DeploymentManagerProvider,
       schedulingForProcessingType: SchedulingForProcessingType,
       deploymentManagerDependencies: DeploymentManagerDependencies,
+      deploymentManagersClassLoader: DeploymentManagersClassLoader,
       engineSetupName: EngineSetupName,
       modelData: ModelData,
       deploymentConfig: Config,
-      metaDataInitializer: MetaDataInitializer,
+      metaDataInitializer: MetaDataInitializer
   ) = {
     val scenarioStateCacheTTL = ScenarioStateCachingConfig.extractScenarioStateCacheTTL(deploymentConfig)
 
     val validDeploymentManager = for {
-      deploymentManager <- deploymentManagerProvider.createDeploymentManager(
-        modelData,
-        deploymentManagerDependencies,
-        deploymentConfig,
-        scenarioStateCacheTTL
-      )
+      deploymentManager <-
+        ThreadUtils.withThisAsContextClassLoader(deploymentManagersClassLoader) {
```

Comment:
as we agreed, let's move this code to some deployment manager creator decorator close to the service load code